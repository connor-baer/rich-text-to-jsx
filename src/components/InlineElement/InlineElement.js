import React, { Fragment } from 'react';

/**
 * Default fallback element for inline nodes.
 * Renders a prominent warning in development.
 */
function InlineElement(props) {
  if (process.env.NODE_ENV !== 'production') {
    // eslint-disable-next-line no-console
    console.debug('Unknown inline node type', props);
    return (
      <Fragment>
        <span style={{ color: 'white', backgroundColor: 'red' }}>
          Unknown inline node type
        </span>
        {props.children}
      </Fragment>
    );
  }
  return props.children || null;
}

if (process.env.NODE_ENV !== 'production') {
  // eslint-disable-next-line global-require
  const PropTypes = require('prop-types');

  InlineElement.propTypes = {
    /**
     * The element children.
     */
    children: PropTypes.oneOfType([PropTypes.element, PropTypes.array]),
  };
}

/**
 * @component
 */
export default InlineElement;
