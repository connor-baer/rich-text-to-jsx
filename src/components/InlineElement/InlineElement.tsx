import React, { Fragment } from 'react';

export interface InlineElementProps {
  /**
   * The element children.
   */
  children?: JSX.Element;
}

/**
 * Default fallback element for inline nodes.
 * Renders a prominent warning in development.
 */
export default function InlineElement(
  props: InlineElementProps,
): JSX.Element | null {
  if (process.env.NODE_ENV !== 'production') {
    // eslint-disable-next-line no-console
    console.debug('Unknown inline node type', props);
    return (
      <Fragment>
        <span style={{ color: 'white', backgroundColor: 'red' }}>
          Unknown inline node type
        </span>
        {props.children}
      </Fragment>
    );
  }
  return props.children || null;
}
