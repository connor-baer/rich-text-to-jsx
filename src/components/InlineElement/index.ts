import InlineElement from './InlineElement';

export default InlineElement;
