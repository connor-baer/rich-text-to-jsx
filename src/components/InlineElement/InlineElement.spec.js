import React from 'react';
import { create } from 'react-test-renderer';

import InlineElement from './InlineElement';

/* eslint-disable no-console */

describe('InlineElement', () => {
  describe('in development', () => {
    const nodeEnv = process.env.NODE_ENV;
    const { debug } = console;

    beforeAll(() => {
      console.debug = jest.fn();
      process.env.NODE_ENV = 'development';
    });

    afterAll(() => {
      console.debug = debug;
      process.env.NODE_ENV = nodeEnv;
    });

    it('should log a debug message to the console', () => {
      create(<InlineElement />);
      expect(console.debug).toHaveBeenCalledTimes(1);
    });

    it('should render a warning', () => {
      const actual = create(<InlineElement />);
      expect(actual).toMatchSnapshot();
    });

    it('should render a warning and the children', () => {
      const actual = create(
        <InlineElement>
          <p>Foo</p>
        </InlineElement>,
      );
      expect(actual).toMatchSnapshot();
    });
  });

  describe('in production', () => {
    const nodeEnv = process.env.NODE_ENV;

    beforeAll(() => {
      process.env.NODE_ENV = 'production';
    });

    afterAll(() => {
      process.env.NODE_ENV = nodeEnv;
    });

    it('should render null', () => {
      const actual = create(<InlineElement />);
      expect(actual).toMatchSnapshot();
    });

    it('should render the children', () => {
      const actual = create(
        <InlineElement>
          <p>Bar</p>
        </InlineElement>,
      );
      expect(actual).toMatchSnapshot();
    });
  });
});
