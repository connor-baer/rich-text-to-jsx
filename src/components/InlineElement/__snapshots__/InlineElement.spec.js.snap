// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`InlineElement in development should render a warning 1`] = `
<span
  style="color: white; background-color: red;"
>
  Unknown inline node type
</span>
`;

exports[`InlineElement in development should render a warning and the children 1`] = `
<div>
  <span
    style="color: white; background-color: red;"
  >
    Unknown inline node type
  </span>
  <p>
    Foo
  </p>
</div>
`;

exports[`InlineElement in production should render null 1`] = `null`;

exports[`InlineElement in production should render the children 1`] = `
<p>
  Bar
</p>
`;
